mlflow/sklearn.py

```diff
@@ -500,3 +505,183 @@ def should_log(self):
+    # packages
+    with ignore_warnings(category=FutureWarning):
+        for _, modname, _ in pkgutil.walk_packages(path=[root], prefix="sklearn."):
+            mod_parts = modname.split(".")
+            if any(part in modules_to_ignore for part in mod_parts) or "._" in modname:
+                continue
+            module = import_module(modname)
+            classes = inspect.getmembers(module, inspect.isclass)
+            classes = [(name, est_cls) for name, est_cls in classes if not name.startswith("_")]
+
+            # TODO: Remove when FeatureHasher is implemented in PYPY
+            # Skips FeatureHasher for PYPY
+            if IS_PYPY and "feature_extraction" in modname:
+                classes = [(name, est_cls) for name, est_cls in classes if name == "FeatureHasher"]
+
+            all_classes.extend(classes)
+
+    all_classes = set(all_classes)
+
+    estimators = [
+        c for c in all_classes if (issubclass(c[1], BaseEstimator) and c[0] != "BaseEstimator")
+    ]
+    # get rid of abstract base classes
+    estimators = [c for c in estimators if not is_abstract(c[1])]
+
+    if type_filter is not None:
+        if not isinstance(type_filter, list):
+            type_filter = [type_filter]
+        else:
+            type_filter = list(type_filter)  # copy
+        filtered_estimators = []
+        filters = {
+            "classifier": ClassifierMixin,
+            "regressor": RegressorMixin,
+            "transformer": TransformerMixin,
+            "cluster": ClusterMixin,
+        }
+        for name, mixin in filters.items():
+            if name in type_filter:
+                type_filter.remove(name)
+                filtered_estimators.extend([est for est in estimators if issubclass(est[1], mixin)])
+        estimators = filtered_estimators
+        if type_filter:
+            raise ValueError(
+                "Parameter type_filter must be 'classifier', "
+                "'regressor', 'transformer', 'cluster' or "
+                "None, got"
+                " %s." % repr(type_filter)
+            )
+
+    # drop duplicates, sort for reproducibility
+    # itemgetter is used to ensure the sort does not extend to the 2nd item of
+    # the tuple
+    return sorted(set(estimators), key=itemgetter(0))
+
+
+def autolog():
+    """
+    Enable autologging for scikit-learn.
+    """
+    import sklearn
+
+    if _is_old_version():
+        warnings.warn(
+            "Autologging utilities may not work properly on scikit-learn < 0.20.3 "
+            "(current version: {})".format(sklearn.__version__)
+        )
+
+    def fit_mlflow(self, func_name, *args, **kwargs):
+        active_run_exists = bool(mlflow.active_run())
+        if not active_run_exists:
+            try_mlflow_log(mlflow.start_run)
+
+        # TODO: We should not log nested estimator parameters for
+        # parameter search estimators (GridSearchCV, RandomizedSearchCV)
+        try_mlflow_log(mlflow.log_params, self.get_params(deep=True))
```

Comment:
In the rare / unlikely case where the model has more than 100 top-level parameter keys or produces more than 1MB in aggregate parameter output, this may fail due to limits of the underlying `Log Batch` API (https://mlflow.org/docs/latest/rest-api.html#log-batch). Can we handle this case by chunking the set of parameters?